src/Relude/Extra/Validation.hs

```diff
@@ -1,51 +1,78 @@
-{-# LANGUAGE CPP        #-}
-{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE CPP                  #-}
+{-# LANGUAGE ConstraintKinds      #-}
+{-# LANGUAGE DataKinds            #-}
+{-# LANGUAGE LambdaCase           #-}
+{-# LANGUAGE TypeFamilies         #-}
+{-# LANGUAGE TypeOperators        #-}
+{-# LANGUAGE UndecidableInstances #-}
 
 {- |
 Copyright:  (c) 2014 Chris Allen, Edward Kmett
             (c) 2018-2019 Kowainik
 SPDX-License-Identifier: MIT
 Maintainer: Kowainik <xrom.xkov@gmail.com>
 
-'Validation' is a monoidal sibling to 'Either'. Use 'Validation' on operations that
-might fail with multiple errors that you want to preserve.
--}
+'Validation' is a monoidal sibling to 'Either' but 'Validation' doesn't have a
+'Monad' instance. 'Validation' allows to accumulate all errors instead of
+short-circuting on the first error so you can display all possible errors at
+once. Common use-cases include:
+
+1. Validating each input a form with multiple inputs.
```

Comment:
"a form" seems to be out of the context

Reply:
Ah, yeah, probably I missed a few words...